components/atg/Editor/components/file-tree.tsx

```diff
@@ -0,0 +1,216 @@
+  );
+};
+
+interface SubTreeProps {
+  directory: Directory;
+  selectedFile: File | undefined;
+  onSelect: (file: File) => void;
+  themeSub: boolean;
+}
+
+const SubTree = (props: SubTreeProps) => {
+  const [emptyDirectory, setEmptyDirectory] = useState<boolean>(false);
+  return (
+    <div>
+      {emptyDirectory ? (
+        <div className="h-2 flex items-center "></div>
+      ) : (
+        <>
+          {props.directory.dirs.sort(sortDir).map((dir) => (
+            <DirDiv
+              key={dir.id}
+              directory={dir}
+              selectedFile={props.selectedFile}
+              onSelect={props.onSelect}
+              themeDir={props.themeSub}
+            />
+          ))}
+          {props.directory.files.sort(sortFile).map((file) => (
+            <FileDiv
+              key={file.id}
+              file={file}
+              selectedFile={props.selectedFile}
+              onClick={() => props.onSelect(file)}
+              themeFileDiv={props.themeSub}
+            />
+          ))}
+        </>
+      )}
+    </div>
+  );
+};
+
+const FileDiv = ({
+  file,
+  icon,
+  selectedFile,
+  onClick,
+  openParameter,
+  themeFileDiv,
+}: {
+  file: File | Directory;
+  icon?: string;
+  selectedFile: File | undefined;
+  onClick: () => void;
+  openParameter?: boolean;
+  themeFileDiv: boolean;
+}) => {
+  const isSelected = selectedFile && selectedFile.id === file.id;
+  const depth = file.depth;
+
+  return (
+    <div
+      className={`flex items-center ${
+        isSelected
+          ? `${
+              themeFileDiv
+                ? "bg-gray-200 border-y-1 border-y-gray-400"
+                : "bg-gray-700"
+            }`
+          : "bg-transparent"
+      } cursor-pointer ${
+        themeFileDiv
+          ? !isSelected
+            ? "hover:bg-gray-200"
+            : "hover:bg-gray-300"
+          : !isSelected
+          ? "hover:bg-gray-800"
+          : "hover:bg-gray-600"
+      } `}
+      onClick={onClick}
+      style={{ paddingLeft: `${(depth + 1) * 16}px` }} // Adjust padding for depth, add an additional level for files
+    >
+      {openParameter !== undefined &&
+        (openParameter ? (
+          <MdOutlineKeyboardArrowDown />
+        ) : (
+          <MdOutlineKeyboardArrowRight />
+        ))}
+      <FileIcon
+        name={icon}
+        extension={file?.name?.split(".").pop() || ""}
+        themeIcons={themeFileDiv}
+      />
+      <span className="ml-1">{file.name}</span>
+    </div>
+  );
+};
+
+const DirDiv = ({
+  directory,
+  selectedFile,
+  onSelect,
+  themeDir,
+}: {
+  directory: Directory;
+  selectedFile: File | undefined;
+  onSelect: (file: File) => void;
+  themeDir: boolean;
+}) => {
+  const [open, setOpen] = useState(() =>
+    isChildSelected(directory, selectedFile)
+  );
+  const [dirState, setDirState] = useState(directory);
+  useState(() => {
+    if (
+      directory.id === "src" ||
+      directory.id === "src2" ||
```

Comment:
add comment for why is this done

Reply:
added the comment for it. Basically it was for when the file tree renders it's top directory which is src. It should render open meaning all of the files will be displayed rather than user going all the way to click a button and then the file-tree would open.